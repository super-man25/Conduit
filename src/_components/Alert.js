// @flow
import * as React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import styled from 'styled-components';

import { colors, shadows, zIndexes } from '_constants';
import { actions } from '_state/alert';
import { Icon } from './Icon';
import { Flex } from './Flex';

const StyledApiAlert: React.ComponentType<{}> = styled(Flex)`
  position: absolute;
  right: 25px;
  top: 25px;
  z-index: ${zIndexes.BASE};
  width: 350px;
  background-color: ${colors.white};
  border-color: ${(props) =>
    props.type === 'apiError' ? colors.red : colors.green};
  box-shadow: ${shadows.SMALL};
  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
  cursor: pointer;
  padding: 25px;
`;

const StatusIcon: React.ComponentType<{}> = styled.div`
  margin-right: 15px;
`;

const StatusText: React.ComponentType<{}> = styled.div`
  font-size: 20px;
  font-weight: 300;
  letter-spacing: 1px;
  color: ${(props) => props.color};
  margin-bottom: 3px;
`;

const MsgText: React.ComponentType<{}> = styled.div`
  font-size: 13px;
  font-weight: 300;
`;

export const Alert = () => {
  const alertState = useSelector(({ alert }) => alert);
  const dispatch = useDispatch();
  const clearAlert = () => dispatch(actions.clear());

  const show = alertState.type !== null && alertState.message !== null;
  const statusColor =
    alertState.type === 'apiError' ? colors.red : colors.green;
  const statusText = alertState.type === 'apiError' ? 'Error:' : 'Success!';

  return (
    <StyledApiAlert {...alertState} show={show} onClick={clearAlert}>
      <StatusIcon>
        <Icon name={alertState.type} size={40} color={statusColor} />
      </StatusIcon>
      <Flex direction="column">
        <StatusText color={statusColor}>{statusText}</StatusText>
        <MsgText>{alertState.message}</MsgText>
      </Flex>
    </StyledApiAlert>
  );
};
