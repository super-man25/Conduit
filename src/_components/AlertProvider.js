import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { history } from '_helpers';
import { actions as alertActions } from '_state/alert';

class AlertProviderPresenter extends React.Component {
  constructor(props) {
    super(props);

    const { dispatch } = this.props;
    history.listen(() => {
      dispatch(alertActions.clear());
    });
  }

  render() {
    const { alert, children } = this.props;

    // TODO: style alerts
    return (
      <div>
        {alert &&
          alert.show && (
            <div className={`alert ${alert.type}`}>{alert.message}</div>
          )}
        {children}
      </div>
    );
  }
}

AlertProviderPresenter.propTypes = {
  /** Redux dispatch */
  dispatch: PropTypes.func,

  /** Alert object */
  alert: PropTypes.shape({
    message: PropTypes.string,
    type: PropTypes.string
  }),

  /** Children */
  children: PropTypes.node
};

function mapStateToProps(state) {
  return {
    alertState: state.alert
  };
}

export const AlertProvider = connect(mapStateToProps)(AlertProviderPresenter);
