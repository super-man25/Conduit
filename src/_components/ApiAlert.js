// @flow

import * as React from 'react';
import { connect } from 'react-redux';
import styled from 'styled-components';
import { cssConstants, shadows, zIndexes } from '_constants';
import { Icon } from './Icon';

type Props = {
  alertState: {
    type: 'api-error' | 'api-success',
    message: string
  }
};

export const ApiAlertDiv: React.ComponentType<{}> = styled.div`
  position: absolute;
  right: 24px;
  top: 24px;
  z-index: ${zIndexes.BASE};
  font-size: 18px;
  width: 357px;
  height: 90px;
  background-color: ${cssConstants.PRIMARY_WHITE};
  border-color: ${(props) =>
    props.type === 'api-error'
      ? cssConstants.SECONDARY_RED
      : cssConstants.SECONDARY_GREEN};
  box-shadow: ${shadows.SMALL};
  visibility: ${(props) => (props.show ? 'visible' : 'hidden')};
`;

const StatusIcon: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 26px;
  left: 26px;
`;

const StatusText: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 26px;
  left: 93px;
  font-size: 1.2em;
  font-weight: 300;
  letter-spacing: 1px;
  color: ${(props) => props.color};
`;

const MsgText: React.ComponentType<{}> = styled.div`
  position: absolute;
  top: 54px;
  left: 93px;
  font-size: 0.7em;
  font-weight: 300;
  margin-right: 24px;
  color: ${cssConstants.PRIMARY_BLACK};
`;

export class ApiAlertPresenter extends React.Component<Props> {
  render() {
    const { alertState } = this.props;
    const show = {
      show: alertState.type !== null && alertState.message !== null
    };
    const statusColor =
      alertState.type === 'api-error'
        ? cssConstants.SECONDARY_RED
        : cssConstants.SECONDARY_GREEN;
    const statusText = alertState.type === 'api-error' ? 'Error:' : 'Success!';

    return (
      <ApiAlertDiv {...alertState} {...show}>
        <StatusIcon>
          <Icon name={alertState.type} size={39} color={statusColor} />
        </StatusIcon>
        <StatusText color={statusColor}>{statusText}</StatusText>
        <MsgText>{alertState.message}</MsgText>
      </ApiAlertDiv>
    );
  }
}

function mapStateToProps(state) {
  return {
    alertState: state.alert
  };
}

export const ApiAlert = connect(
  mapStateToProps,
  null
)(ApiAlertPresenter);
