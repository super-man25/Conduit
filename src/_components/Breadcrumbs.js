import styled from 'styled-components';
import { cssConstants } from '../_constants';

export const Breadcrumbs = styled.div`
  width: 100%;
  height: 30px;
  margin: 0;
  padding: 0;
  background: ${cssConstants.PRIMARY_WHITE};
  font-size: 12px;
`;
