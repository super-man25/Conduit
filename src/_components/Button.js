// @flow

import * as React from 'react';
import styled from 'styled-components';
import { cssConstants, shadows } from '_constants';
import { withBoxModelProps } from '_helpers/style-utils';

type Props = {
  fontSize: string,
  textAlign: string,
  minWidth: string,
};

export const Button: React.ComponentType<Props> = withBoxModelProps(styled.button`
  border: none;
  min-width: ${(props) => props.minWidth || '100px'};
  padding: 1rem 2rem;
  text-align: ${(props) => props.textAlign || 'center'};
  border-radius: 3px;
  font-size: ${(props) => props.fontSize || '1rem'};
  cursor: pointer;
  transition-duration: 0.3s;

  &:disabled {
    cursor: not-allowed;
  }
`);

export const PrimaryButton = styled(Button)`
  transition: opacity 100ms ease-out;
  color: ${cssConstants.PRIMARY_WHITE};
  background-color: ${cssConstants.PRIMARY_BLUE};

  &:disabled {
    color: ${cssConstants.PRIMARY_GRAY};
    background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
    opacity: 0.5;
  }

  &:active {
    background-color: ${cssConstants.PRIMARY_LIGHT_BLUE};
    box-shadow: ${shadows.ACTIVE_BUTTON};
  }

  &:hover:not(:disabled) {
    background-color: ${cssConstants.PRIMARY_BLUE_HOVER};
  }
`;

export const SecondaryButton = styled(Button)`
  background-color: ${cssConstants.PRIMARY_WHITE};
  border: 2px solid ${cssConstants.PRIMARY_BLUE};
  color: ${cssConstants.PRIMARY_BLUE};
  transition: border linear;

  &:disabled {
    border: 2px solid ${cssConstants.PRIMARY_GRAY};
    color: ${cssConstants.PRIMARY_GRAY};
    opacity: 0.5;
  }

  &:active {
    background-color: ${cssConstants.SECONDARY_BACKGROUND_LIGHTEST_BLUE};
    color: ${cssConstants.PRIMARY_LIGHT_BLUE};
    outline: 2px solid ${cssConstants.PRIMARY_LIGHT_BLUE};
    box-shadow: ${shadows.ACTIVE_BUTTON};
  }

  &:hover:not(:disabled) {
    border: 3px solid ${cssConstants.PRIMARY_BLUE};
    background-color: ${cssConstants.SECONDARY_BACKGROUND_LIGHTEST_BLUE};
    color: ${cssConstants.PRIMARY_BLUE};
  }
`;

export const TertiaryButton = styled(Button)`
  background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
  color: ${cssConstants.PRIMARY_LIGHT_BLACK};
  opacity: 0.8;

  &:disabled {
    opacity: 0.5;
  }

  &:hover:not(:disabled) {
    opacity: 1;
    background-color: ${cssConstants.PRIMARY_LIGHT_GRAY};
  }
`;

export const TextButton = styled(Button)`
  color: ${cssConstants.PRIMARY_BLUE};
  background-color: inherit;

  &:disabled {
    color: ${cssConstants.PRIMARY_LIGHT_GRAY};
    opacity: 0.5;
  }

  &:hover:not(:disabled) {
    color: ${cssConstants.PRIMARY_LIGHT_BLUE};
  }
`;
