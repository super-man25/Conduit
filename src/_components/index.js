export * from './Alert';
export * from './AsyncButton';
export * from './Breadcrumbs';
export * from './Button';
export * from './Center';
export * from './CenteredLoader';
export * from './ChartContainer';
export * from './ChartLegend';
export * from './Checkbox';
export * from './ChipButton';
export * from './ContentWrapper';
export * from './DatePicker';
export * from './DateRangeDropdown';
export * from './DateRangeFilter';
export * from './Dropdown';
export * from './DropdownButton';
export * from './EDLink';
export * from './EDText';
export * from './Flex';
export * from './Form';
export * from './FormInput';
export * from './FormInputArray';
export * from './HelpBlockDiv';
export * from './HorizontalBars';
export * from './Icon';
export * from './Icon/SVGIcons';
export * from './ImageLayout';
export * from './Input';
export * from './Integration';
export * from './Layout';
export * from './LeftNav';
export * from './LegendItem';
export * from './Loader';
export * from './MailtoLink';
export * from './MainContent';
export * from './Modal';
export * from './Panel';
export * from './ReportDownloadButton';
export * from './ScreenTitleBlock';
export * from './ScrollableList';
export * from './SecuredRoute';
export * from './SelectBox';
export * from './SelectDropdown';
export * from './Setting';
export * from './SettingButton';
export * from './SiteHeader';
export * from './SortableButton';
export * from './Spacing';
export * from './StyledTags';
export * from './Table';
export * from './Toggle';
export * from './UserWelcome';
