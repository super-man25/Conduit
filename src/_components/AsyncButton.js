// @flow

import * as React from 'react';
import { cssConstants } from '_constants';
import { PrimaryButton, Loader } from '_components';

type Props = {
  isLoading: boolean,
  disabled?: boolean,
  children: React.Node,
};

export const AsyncButton = (props: Props) => {
  const { isLoading, children, ...rest } = props;
  return (
    <PrimaryButton {...rest}>
      {props.isLoading ? (
        <Loader small color={cssConstants.PRIMARY_WHITE} />
      ) : (
        props.children
      )}
    </PrimaryButton>
  );
};
