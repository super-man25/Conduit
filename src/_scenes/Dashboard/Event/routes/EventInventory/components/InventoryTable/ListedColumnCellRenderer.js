// @flow
import * as React from 'react';
import { Toggle, Box } from '_components';
import { connect } from 'react-redux';
import { actions } from '_state/eventInventory';

type Props = {
  cellData: any,
  setListed: () => void
};

export const ListedColumnCellPresenter = ({ cellData, setListed }: Props) => (
  <Box marginLeft="1.25rem">
    <Toggle isChecked={cellData} onChange={setListed} size="xsmall" />
  </Box>
);

const mapDispatchToProps = (dispatch, ownProps) => ({
  setListed: () =>
    dispatch(actions.setEventRowListed(ownProps.rowData, !ownProps.cellData))
});

const ListedColumnCell = connect(
  null,
  mapDispatchToProps
)(ListedColumnCellPresenter);

export const listedColumnCellRenderer = (props: Props) => (
  <ListedColumnCell {...props} />
);
