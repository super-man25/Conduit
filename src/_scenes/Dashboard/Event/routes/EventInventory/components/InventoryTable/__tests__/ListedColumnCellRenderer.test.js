import React from 'react';
import { shallow } from 'enzyme';
import { ListedColumnCellPresenter } from '../ListedColumnCellRenderer';
import { Toggle } from '_components';

describe('<ListedColumnCellPresenter />', () => {
  const props = {
    cellData: 'Data',
    updateEditedRowProperty: jest.fn(),
    editedRowState: {
      isListed: true
    }
  };

  it('should render correctly', () => {
    const wrapper = shallow(<ListedColumnCellPresenter {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setListed when the Toggles onChange is triggered', () => {
    const clonedProps = { ...props };
    const wrapper = shallow(<ListedColumnCellPresenter {...clonedProps} />);
    wrapper
      .find(Toggle)
      .first()
      .prop('onChange')();

    expect(clonedProps.updateEditedRowProperty).toHaveBeenCalled();
  });
});
