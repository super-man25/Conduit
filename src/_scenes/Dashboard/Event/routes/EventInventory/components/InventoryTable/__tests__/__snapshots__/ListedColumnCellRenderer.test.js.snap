// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListedColumnCellPresenter /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    cellData="Data"
    setListed={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Toggle
        isChecked="Data"
        onChange={[Function]}
        size="xsmall"
/>,
      "marginLeft": "1.25rem",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "isChecked": "Data",
        "onChange": [Function],
        "size": "xsmall",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Toggle
          isChecked="Data"
          onChange={[Function]}
          size="xsmall"
/>,
        "marginLeft": "1.25rem",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "isChecked": "Data",
          "onChange": [Function],
          "size": "xsmall",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
